File: streampark-common/src/main/scala/org/apache/streampark/common/conf/CommonConfig.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.streampark.common.conf

import java.lang.{Integer => JavaInt, Long => JavaLong}

object CommonConfig {

  val STREAMPARK_WORKSPACE_LOCAL: InternalOption = InternalOption(
    key = "streampark.workspace.local",
    defaultValue = "/streampark",
    classType = classOf[String])

  val STREAMPARK_WORKSPACE_REMOTE: InternalOption = InternalOption(
    key = "streampark.workspace.remote",
    defaultValue = "/streampark",
    classType = classOf[String])

  val STREAMPARK_HADOOP_USER_NAME: InternalOption = InternalOption(
    key = "streampark.hadoop-user-name",
    defaultValue = "hdfs",
    classType = classOf[String])

  val STREAMPARK_PROXY_YARN_URL: InternalOption = InternalOption(
    key = "streampark.proxy.yarn-url",
    defaultValue = "",
    classType = classOf[String],
    description = "proxy yarn url. ex: knox proxy or other")

  val STREAMPARK_YARN_AUTH: InternalOption = InternalOption(
    key = "streampark.yarn.http-auth",
    defaultValue = "",
    classType = classOf[String],
    description = "yarn http auth type. ex: simple, kerberos")

  val DOCKER_HOST: InternalOption = InternalOption(
    key = "streampark.docker.http-client.docker-host",
    defaultValue = "",
    classType = classOf[String],
    description = "docker host for DockerHttpClient")

  val DOCKER_MAX_CONNECTIONS: InternalOption = InternalOption(
    key = "streampark.docker.http-client.max-connections",
    defaultValue = 100,
    classType = classOf[JavaInt],
    description = "instantiating max connections for DockerHttpClient")

  val DOCKER_CONNECTION_TIMEOUT_SEC: InternalOption = InternalOption(
    key = "streampark.docker.http-client.connection-timeout-sec",
    defaultValue = 100L,
    classType = classOf[JavaLong],
    description = "instantiating connection timeout for DockerHttpClient")

  val DOCKER_RESPONSE_TIMEOUT_SEC: InternalOption = InternalOption(
    key = "streampark.docker.http-client.response-timeout-sec",
    defaultValue = 120L,
    classType = classOf[JavaLong],
    description = "instantiating connection timeout for DockerHttpClient")

  val MAVEN_SETTINGS_PATH: InternalOption = InternalOption(
    key = "streampark.maven.settings",
    defaultValue = null,
    classType = classOf[String],
    description = "maven settings.xml full path")

  val MAVEN_REMOTE_URL: InternalOption = InternalOption(
    key = "streampark.maven.central.repository",
    defaultValue = "https://repo1.maven.org/maven2/",
    classType = classOf[String],
    description = "maven repository used for built-in compilation")

  val MAVEN_AUTH_USER: InternalOption = InternalOption(
    key = "streampark.maven.auth.user",
    defaultValue = null,
    classType = classOf[String],
    description = "maven repository used for built-in compilation")

  val MAVEN_AUTH_PASSWORD: InternalOption = InternalOption(
    key = "streampark.maven.auth.password",
    defaultValue = null,
    classType = classOf[String],
    description = "maven repository used for built-in compilation")

  val KERBEROS_TTL: InternalOption = InternalOption(
    key = "security.kerberos.ttl",
    defaultValue = "2h",
    classType = classOf[String],
    description = "kerberos default ttl")

  val READ_LOG_MAX_SIZE: InternalOption = InternalOption(
    key = "streampark.read-log.max-size",
    defaultValue = "1mb",
    classType = classOf[String],
    description = "The maximum size of the default read log")

  val SPRING_PROFILES_ACTIVE: InternalOption = InternalOption(
    key = "spring.profiles.active",
    defaultValue = "h2",
    classType = classOf[String],
    description = "Use the database type")

}

File: streampark-common/src/main/java/org/apache/streampark/common/conf/CommonConfig.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.streampark.common.conf;

/** Common internal configuration options. */
public final class CommonConfig {

    private CommonConfig() {
    }

    private static final InternalOption STREAMPARK_WORKSPACE_LOCAL_OPTION =
        new InternalOption("streampark.workspace.local", "/streampark", String.class);

    private static final InternalOption STREAMPARK_WORKSPACE_REMOTE_OPTION =
        new InternalOption("streampark.workspace.remote", "/streampark", String.class);

    private static final InternalOption STREAMPARK_HADOOP_USER_NAME_OPTION =
        new InternalOption("streampark.hadoop-user-name", "hdfs", String.class);

    private static final InternalOption STREAMPARK_PROXY_YARN_URL_OPTION =
        new InternalOption(
            "streampark.proxy.yarn-url",
            "",
            String.class,
            "proxy yarn url. ex: knox proxy or other");

    private static final InternalOption STREAMPARK_YARN_AUTH_OPTION =
        new InternalOption(
            "streampark.yarn.http-auth",
            "",
            String.class,
            "yarn http auth type. ex: simple, kerberos");

    private static final InternalOption DOCKER_HOST_OPTION =
        new InternalOption(
            "streampark.docker.http-client.docker-host",
            "",
            String.class,
            "docker host for DockerHttpClient");

    private static final InternalOption DOCKER_MAX_CONNECTIONS_OPTION =
        new InternalOption(
            "streampark.docker.http-client.max-connections",
            100,
            Integer.class,
            "instantiating max connections for DockerHttpClient");

    private static final InternalOption DOCKER_CONNECTION_TIMEOUT_SEC_OPTION =
        new InternalOption(
            "streampark.docker.http-client.connection-timeout-sec",
            100L,
            Long.class,
            "instantiating connection timeout for DockerHttpClient");

    private static final InternalOption DOCKER_RESPONSE_TIMEOUT_SEC_OPTION =
        new InternalOption(
            "streampark.docker.http-client.response-timeout-sec",
            120L,
            Long.class,
            "instantiating connection timeout for DockerHttpClient");

    private static final InternalOption MAVEN_SETTINGS_PATH_OPTION =
        new InternalOption(
            "streampark.maven.settings",
            null,
            String.class,
            "maven settings.xml full path");

    private static final InternalOption MAVEN_REMOTE_URL_OPTION =
        new InternalOption(
            "streampark.maven.central.repository",
            "https://repo1.maven.org/maven2/",
            String.class,
            "maven repository used for built-in compilation");

    private static final InternalOption MAVEN_AUTH_USER_OPTION =
        new InternalOption(
            "streampark.maven.auth.user",
            null,
            String.class,
            "maven repository used for built-in compilation");

    private static final InternalOption MAVEN_AUTH_PASSWORD_OPTION =
        new InternalOption(
            "streampark.maven.auth.password",
            null,
            String.class,
            "maven repository used for built-in compilation");

    private static final InternalOption KERBEROS_TTL_OPTION =
        new InternalOption("security.kerberos.ttl", "2h", String.class, "kerberos default ttl");

    private static final InternalOption READ_LOG_MAX_SIZE_OPTION =
        new InternalOption(
            "streampark.read-log.max-size",
            "1mb",
            String.class,
            "The maximum size of the default read log");

    private static final InternalOption SPRING_PROFILES_ACTIVE_OPTION =
        new InternalOption(
            "spring.profiles.active", "h2", String.class, "Use the database type");

    public static InternalOption STREAMPARK_WORKSPACE_LOCAL() {
        return STREAMPARK_WORKSPACE_LOCAL_OPTION;
    }

    public static InternalOption STREAMPARK_WORKSPACE_REMOTE() {
        return STREAMPARK_WORKSPACE_REMOTE_OPTION;
    }

    public static InternalOption STREAMPARK_HADOOP_USER_NAME() {
        return STREAMPARK_HADOOP_USER_NAME_OPTION;
    }

    public static InternalOption STREAMPARK_PROXY_YARN_URL() {
        return STREAMPARK_PROXY_YARN_URL_OPTION;
    }

    public static InternalOption STREAMPARK_YARN_AUTH() {
        return STREAMPARK_YARN_AUTH_OPTION;
    }

    public static InternalOption DOCKER_HOST() {
        return DOCKER_HOST_OPTION;
    }

    public static InternalOption DOCKER_MAX_CONNECTIONS() {
        return DOCKER_MAX_CONNECTIONS_OPTION;
    }

    public static InternalOption DOCKER_CONNECTION_TIMEOUT_SEC() {
        return DOCKER_CONNECTION_TIMEOUT_SEC_OPTION;
    }

    public static InternalOption DOCKER_RESPONSE_TIMEOUT_SEC() {
        return DOCKER_RESPONSE_TIMEOUT_SEC_OPTION;
    }

    public static InternalOption MAVEN_SETTINGS_PATH() {
        return MAVEN_SETTINGS_PATH_OPTION;
    }

    public static InternalOption MAVEN_REMOTE_URL() {
        return MAVEN_REMOTE_URL_OPTION;
    }

    public static InternalOption MAVEN_AUTH_USER() {
        return MAVEN_AUTH_USER_OPTION;
    }

    public static InternalOption MAVEN_AUTH_PASSWORD() {
        return MAVEN_AUTH_PASSWORD_OPTION;
    }

    public static InternalOption KERBEROS_TTL() {
        return KERBEROS_TTL_OPTION;
    }

    public static InternalOption READ_LOG_MAX_SIZE() {
        return READ_LOG_MAX_SIZE_OPTION;
    }

    public static InternalOption SPRING_PROFILES_ACTIVE() {
        return SPRING_PROFILES_ACTIVE_OPTION;
    }
}
